GameServerLib/Game.cs

```diff
@@ -110,14 +117,54 @@ public void Initialize(ushort port, string blowfishKey, Config config)
             _pauseTimer.Elapsed += (sender, args) => PauseTimeLeft--;
             PauseTimeLeft = 30 * 60; // 30 minutes
 
+            // TODO: GameApp should send the Response/Request handlers
             _packetServer = new PacketServer();
-            _packetServer.InitServer(port, blowfishKey, this);
+            _packetServer.InitServer(port, blowfishKey, this, RequestHandler, ResponseHandler);
+            // TODO: switch the notifier with ResponseHandler
             PacketNotifier = new PacketNotifier(_packetServer.PacketHandlerManager, Map.NavGrid);
-            // TODO: make lib to only get API types and not byte[], start from removing this line
-            PacketReader = new PacketReader();
+            InitializePacketHandlers();
 
             _logger.Info("Game is ready.");
         }
+        public void InitializePacketHandlers()
+        {
+            // maybe use reflection, the problem is that Register is generic and so it needs to know its type at 
+            // compile time, maybe just use interface and in runetime figure out the type - and again there is
+            // a problem with passing generic delegate to non-generic function, if we try to only constraint the
+            // argument to interface ICoreRequest we will get an error cause our generic handlers use generic type
+            // even with where statement that doesn't work
+            RequestHandler.Register<AttentionPingRequest>(new HandleAttentionPing(this).HandlePacket);
```

Comment:
Not so sure about this maybe have ``RequestHandler.Register<AttentionPingRequest, HandleAttentionPing>();`` instead?

Reply:
the generic thing get the type of the request as type, and the function that handles the request as variable, so I don't think you can do that like how you propose.

Reply:
You could however pass the class to it

Reply:
Im not sure, cause this generic thing will need to know that there is HandlePacket method with exactly the same type, can check this.